code/script.js

```diff
@@ -249,95 +251,147 @@ const setSecret = () => {
 const filterCharacters = (keep) => {
   // Show the correct alert message for different categories
-  if (group === 'accessories') {
+  const attribute = currentQuestion.attribute
+  const value = currentQuestion.value
+  const category = currentQuestion.category
+
+  if (category === 'accessories') {
     if (keep) {
-      alert(
-        `Yes, the person wears ${attribute}! Keep all that wears ${attribute}`
-      )
+      alert(`Yes, the person wears ${attribute}! Keep all that wears ${attribute}`)
     } else {
-      alert(
-        `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`
-      )
+      alert(`No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`)
     }
-  } else if (group === 'other') {
-    // Similar to the one above
-  } else {
+  } else if (category === 'other') {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
+      alert(`Yes, the person is ${attribute}! Keep all that are ${attribute}'s`)
     } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
+      alert(`No, the person is not ${attribute}! Remove all ${attribute}'s`)
+    }
+  } else if (category === 'hair color') {
+    if (keep) {
+      alert(`Yes, the person has ${value} hair! Keeping all the people with ${value} hair.`)
+    } else {
+      alert(`No, the person doesn't have ${value} hair. Removing people with ${value} hair.`)
+    }
+  } else if (category === 'eye color')
+    if (keep) {
+      alert(`Yes, the person has ${value} eyes! Keeping all the people with ${value} eyes.`)
+    } else {
+      alert(`No, the person doesn't have ${value} eyes. Removing people with ${value} eyes.`)
     }
-  }
 
   // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
-
+  if (keep) {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
+  } else {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  }
   // Invoke a function to redraw the board with the remaining people.
+  generateBoard()
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (suspect) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
+   // store the interaction from the player in a variable.
+  const confirmSuspect = confirm(`Are you sure you want that your guess is ${suspect}?`)
   // If the player wants to guess, invoke the checkMyGuess function.
+  if (confirmSuspect === true) {
+    checkMyGuess(suspect)
+  }
 }
 
 // If you confirm, this function is invoked
 const checkMyGuess = (suspect) => {
```

Comment:
Nice and sleek function :) It's super easy to read and to understand and done in very few lines!